### PR TITLE
client: add detailed comment  and revise cancel

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: Song Gao <disxiaofei@163.com>

### What problem does this PR solve?

add detailed comment and revise cancel in `tsLoop`.

### Release note

* No release note

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -321,7 +325,14 @@ func (c *client) tsLoop() {
 						opts     []opentracing.StartSpanOption
 						requests = make([]*tsoRequest, maxMergeTSORequests+1)
 					)
+					defer func() {
+						if cancel != nil {
+							cancel()
+						}
+					}()
 					for {
+						// if the tso stream for the corresponding dcLocation pdclient is not correspond,
```
Comment (JmPotato) [inline client/client.go:334]:
```suggestion
						// If the tso stream for the corresponding dc-location has not been created yet or needs to be re-created,
```

File: client/client.go [hunk 2]
```diff
@@ -312,6 +312,10 @@ func (c *client) tsLoop() {
 			if !c.checkTSODispatcher(dcLocation) {
 				c.createTSODispatcher(dcLocation)
 				dispatcher, _ := c.tsoDispatcher.Load(dcLocation)
+				// Each goroutine is responsible for handling the tso stream request for its dcLocation.
```
Comment (JmPotato) [inline client/client.go:315]:
```suggestion
				// Each goroutine is responsible for handling the tso stream request for its dc-location.
```

**Review by JmPotato — APPROVED** [review]

File: client/client.go [hunk 3]
```diff
@@ -373,26 +379,23 @@ func (c *client) tsLoop() {
...
 						if err != nil {
 							select {
 							case <-loopCtx.Done():
-								cancel()
 								return
 							default:
 							}
 							log.Error("[pd] getTS error", zap.String("dc-location", dc), errs.ZapError(errs.ErrClientGetTSO, err))
 							c.ScheduleCheckLeader()
-							cancel()
```
Comment (JmPotato) [inline client/client.go:395]: I think this 'cancel()' should be retained.

Comment (Yisaer) [reply client/client.go]: updated.

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**